### PR TITLE
New aria complete button without new tokens

### PR DESCRIPTION
**Pull request by salamaashoush**
1. Use react aria which come with one major change is `onPress` instead of `onClick` read more about why https://react-spectrum.adobe.com/blog/building-a-button-part-1.html. we added `onClick` to allow easy migration but it is deprecated and we should not use it for new code.
2. Unify the `IconButton` and `Button` components using `<Button icon={<SomeIcon/>} />` instead of `IconButton`.
3. Change some props names to be consistent with react-aria naming `compact` -> `isCompact`.
4. `Button` is not a polymorphic component anymore we will have a separate link component.
5. `iconAlign` is replaced by specific icon renders props `startIcon` and `endIcon`, and `icon` is repurposed for icon-only buttons aka `IconButton`.
6. Use `recipe` instead of individual `styleVariants`. 
7. `color` and `variant` are now one prop `variant` and all `alternative` variants are added as a standalone variant eg `<Button color="primary" variant="alternative" />` is now `<Button variant="primaryInverted" />` we use `inverted` postfix is used instead of  `alternative` to match the intended color inversion behavior.
8. The `isCompact` variant now works for all button variations before it only worked for some color variants.

### CODE REVIEW

File: packages/libs/react-ui/src/components/Button/NewButton.css.ts [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+      position: 'relative',
+      display: 'flex',
+      placeItems: 'center',
+      cursor: 'pointer',
+      border: 'none',
+      textDecoration: 'none',
+      borderRadius: tokens.kda.foundation.radius.sm,
+      gap: tokens.kda.foundation.size.n2,
+      paddingInline: tokens.kda.foundation.size.n4,
+      paddingBlock: tokens.kda.foundation.size.n2,
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Button/NewButton.css.ts:156]: can we use atoms for these?

File: packages/libs/react-ui/src/components/Button/NewButton.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  MouseEvent,
+  ReactNode,
+} from 'react';
+import React, { forwardRef, useCallback } from 'react';
+import type { AriaButtonProps } from 'react-aria';
+import { useButton } from 'react-aria';
+import { Loading } from '../Icon/System/SystemIcon';
+import { button, spinner } from './NewButton.css';
+
+type Variants = Omit<NonNullable<RecipeVariants<typeof button>>, 'onlyIcon'>;
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Button/NewButton.tsx:17]: Does this make all recipe variants required props? If so, I don't think we should remove `NonNullable` and have the variants default to what is set in the recipe so users are not required to add every prop

File: packages/libs/react-ui/src/components/Button/NewButton.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+export interface IButtonProps extends AriaButtonProps, Variants {
+  startIcon?: ReactNode;
+  endIcon?: ReactNode;
+  icon?: ReactNode;
+  title?: string;
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Button/NewButton.tsx:22]: can we remove this? I believe people can use `aria-label` for this

**Review by eileenmguo — COMMENTED** [review]
great documentation 🤩

File: packages/libs/react-ui/src/components/Button/NewButton.tsx (hunk 2 above)
Comment (salamaashoush) [reply packages/libs/react-ui/src/components/Button/NewButton.tsx:17]: no it doesn't, it is a workaround for vanilla extract types to make it work with `Omit`

File: packages/libs/react-ui/src/components/Button/NewButton.tsx (hunk 3 above)
Comment (salamaashoush) [reply packages/libs/react-ui/src/components/Button/NewButton.tsx:22]: this is a different thing, this is the native `title` prop which is used to add a native tooltip and it is useful for icon buttons

File: packages/libs/react-ui/src/components/Button/NewButton.css.ts [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+});
+
+export const spinner = style({
+  animationName: rotate,
+  animationDuration: '1.5s',
+  animationIterationCount: 'infinite',
+  animationTimingFunction: 'linear',
+});
+
+export const cloak = style({
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Button/NewButton.css.ts:272]: can be done with atoms

File: packages/libs/react-ui/src/styles/atoms.css.ts [hunk 5]
```diff
@@ -89,22 +90,36 @@ const responsiveProperties = defineProperties({
...
+    padding: [
+      'paddingBlockStart',
+      'paddingBlockEnd',
+      'paddingInlineStart',
+      'paddingInlineEnd',
+    ],
+    paddingX: ['paddingInlineStart', 'paddingInlineEnd'],
+    paddingY: ['paddingBlockStart', 'paddingBlockEnd'],
+    paddingInline: ['paddingInlineStart', 'paddingInlineEnd'],
+    paddingBlock: ['paddingBlockStart', 'paddingBlockEnd'],
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/styles/atoms.css.ts:122]: 😍

File: packages/libs/react-ui/src/components/Button/NewButton.css.ts (hunk 4 above)
Comment (salamaashoush) [reply packages/libs/react-ui/src/components/Button/NewButton.css.ts:272]: This is not needed anymore

File: packages/libs/react-ui/src/styles/atoms.css.ts [hunk 6]
```diff
@@ -3,6 +3,11 @@ import mapValues from 'lodash.mapvalues';
 import { breakpoints, flattenTokens } from './themeUtils';
 import { tokens } from './tokens/contract.css';
 
+export const colorAtoms = flattenTokens({
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/styles/atoms.css.ts:6]: should we add "currentColor" here?

File: packages/libs/react-ui/src/components/ProgressCircle/ProgressCircle.tsx [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+      )}
+      width={spinnerSize}
+      height={spinnerSize}
+      viewBox={`0 0 ${spinnerSize} ${spinnerSize}`}
+      fill="none"
+      strokeWidth={strokeWidth}
+    >
+      <circle role="presentation" cx={center} cy={center} r={r} stroke="gray" />
+      <circle
+        {...testProps(props, 'progress')}
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/ProgressCircle/ProgressCircle.tsx:62]: what is the reason for adding this? I thought we would be testing based on role?

**Review by eileenmguo — COMMENTED** [review]
Looks great!

**Pull request merged**